src/app/dashboard/freelancer/page.tsx

```diff
@@ -27,56 +27,114 @@ import DropdownProfile from '@/components/shared/DropdownProfile';
 import { Skeleton } from '@/components/ui/skeleton';
 import { Button } from '@/components/ui/button';
 import MeetingDialog from '@/components/ui/meetingDialog'; // Import MeetingDialog
+import { ProjectStatus } from '@/utils/freelancer/enum';
 
 interface Project {
   _id: string;
   projectName: string;
+  projectDomain: string[];
   description: string;
+  companyId: string;
   email: string;
+  url?: { value: string }[];
   verified?: any;
   isVerified?: string;
   companyName: string;
   start?: Date;
-  end?: Date;
+  end?: Date | null;
   skillsRequired: string[];
   experience?: string;
-  role: string;
-  projectType: string;
-  totalNeedOfFreelancer?: {
-    category?: string;
-    needOfFreelancer?: number;
-    appliedCandidates?: string[];
-    rejected?: string[];
-    accepted?: string[];
-    status?: string;
+  role?: string;
+  projectType?: string;
+  profiles?: {
+    _id?: string;
+    domain?: string;
+    freelancersRequired?: string;
+    skills?: string[];
+    experience?: number;
+    minConnect?: number;
+    rate?: number;
+    description?: string;
+    domain_id: string;
+    selectedFreelancer?: string[];
+    freelancers?: {
+      freelancerId: string;
+      bidId: string;
+    };
+    totalBid?: string[];
   }[];
-  status?: string;
+  status?: 'Active' | 'Pending' | 'Completed' | 'Rejected';
   team?: string[];
 }
 
 export default function Dashboard() {
   const user = useSelector((state: RootState) => state.user);
   const [projects, setProjects] = useState<Project[]>([]);
-  const [loading, setLoading] = useState(true);
+  const [activeProjects, setActiveProjects] = useState<Project[]>([]);
+  const [pendingProjects, setPendingProjects] = useState<Project[]>([]);
   const [showMeetingDialog, setShowMeetingDialog] = useState(false); // State for showing dialog
+  const [currentTab, setCurrentTab] = useState('Active');
+  const [loading, setLoading] = useState(true);
+  const [loadingStats, setLoadingStats] = useState(true);
 
-  useEffect(() => {
-    const fetchData = async () => {
-      try {
-        const response = await axiosInstance.get(
-          `/freelancer/${user.uid}/project`,
-        );
+  const fetchProjectData = async (status: string) => {
+    setLoading(true);
+    try {
+      const response = await axiosInstance.get(
+        `/freelancer/${user.uid}/project?status=${status}`,
+      );
+      if (response.status == 200 && response?.data?.data) {
         setProjects(response.data.data);
-      } catch (error) {
-        console.error('API Error:', error);
-      } finally {
-        setLoading(false);
       }
-    };
+    } catch (error) {
+      console.error('API Error:', error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const fetchProjectStats = async () => {
+    setLoadingStats(true);
+    try {
+      const activeCountResponse = await axiosInstance.get(
```

Comment:
check if there is an api for fetching project with all status and extract data from there, try to minimize api calls as much as possible

Reply:
we can not seperate this from all projects because it have a big logic in backend for differenciate active pending completed and rejected projects